### DISCUSSION TITLE
How to pass goog:ChromeOptions from maven command line

### DISCUSSION THREAD

**Discussion by sohaib26** [serenity-bdd/serenity-core#3223] [Q&A]
I am trying to pass goog:ChromeOptions from maven command line instead of being dependent on serenity.conf but I am unable to parse it correctly. I have tried in many ways but it is still not working below is once such way I tried:

mvn clean verify serenity:aggregate -Dcucumber.options='src/test/resources/features --tags @Sample --plugin json:target/CucumberReport.json --plugin junit:target/cucumber-reports/Cucumber-Sanity.xml' -Dwebdriver.driver=chrome -Dheadless.mode=true -Dwebdriver.driver.capabilities={'goog:chromeOptions'{args=['no-sandbox', 'disable-dev-shm-usage', 'disable-gpu', 'ignore-certificate-errors']}}

Any help will be much appreciated.

**Comment by wakaleo** [comment]
You won't be able to do that directly from the command line, because the configuration is read from the serenity.conf file in the HOCON format. You can configure different setups using environment sections.

**Reply by sohaib26** [reply]
Thanks. how can I use environment sections in maven command? any reference.

**Comment by sudheer1437** [comment]
> Thanks. how can I use environment sections in maven command? any reference.

This is how i tried, put the below in serenity.conf file and in the command call -Denvironment=firefox or -Denvironment=chrome accordingly

```
environments {
    chrome {
            webdriver {
              driver = chrome
              autodownload=true
              timeouts {
                implicitlywait=60000
                fluentwait=60000
              }
              wait.for.timeout=60000
              capabilities {
                browserName = "chrome"
                acceptInsecureCerts = true
                acceptSslCerts = true
                handlesAlerts = true
                validateTLSCertificates=false
                "goog:chromeOptions" {
                  args = ["start-maximized", "remote-allow-origins=*", "test-type", "no-sandbox", "ignore-certificate-errors", "--window-size=1000,800"
                    "incognito", "disable-infobars", "disable-gpu", "disable-default-apps", "disable-popup-blocking", "disable-extensions-file-access-check",
                    "ignore-ssl-errors=yes", "disable-extensions"],
            #       binary = ${user.dir}\\src\\main\\resources\\windows\\chromedriver.exe
                  prefs {
                    download.default_directory=${user.dir}\\src\\test\\resources\\Downloads
                    download.prompt_for_download = false
                  }
                }
              }
            }
        }

        firefox {
            webdriver {
              driver = firefox
              autodownload=true
              timeouts {
                implicitlywait=60000
                fluentwait=60000
              }
              wait.for.timeout=60000
              capabilities {
                browserName = "firefox"
            #     acceptInsecureCerts = true
            #     acceptSslCerts = true
            #     handlesAlerts = true
            #     validateTLSCertificates=false
            #     "moz:firefoxOptions" {
            #       args = ["start-maximized", "remote-allow-origins=*", "test-type", "no-sandbox", "ignore-certificate-errors", "--window-size=1000,800"
            #                "incognito", "disable-infobars", "disable-gpu", "disable-default-apps", "disable-popup-blocking", "disable-extensions-file-access-check",
            #                "ignore-ssl-errors=yes", "disable-extensions"],
            #       binary = ${user.dir}\\src\\main\\resources\\windows\\geckodriver.exe
            #       prefs {
            #         download.default_directory=${user.dir}\\src\\test\\resources\\Downloads
            #         download.prompt_for_download = false
            #       }
            #     }
              }
            }
        }
	}
```

**Comment by joellis13** [comment]
@wakaleo - Considering the age of this issue (didn't see a more recent one), I'm hoping there may be a change or you may be able to help resolve an issue. We have local, and a couple testing environments configured (they're tiered so we can't swap one for another). We generally do runs in a headless env where we set a window size (and this is what's configured in our serenity.conf) but, because of the tiered environments, we often go into a testing env and run it "locally" in a headed browser.

This means we have to change the configuration in our serenity.conf and would love to, instead, be able to pass at least window-size as a command line option so we can vary with runs. Is this still not possible? If not, can you recommend a way to get around our obstacle and allow us to dynamically configure these properties?

Thanks!
